pkg/controllers/cassandra/util/util.go

```diff
@@ -66,6 +68,35 @@ func SelectorForCluster(c *v1alpha1.CassandraCluster) (labels.Selector, error) {
 	return labels.NewSelector().Add(*clusterNameReq), nil
 }
 
+func SelectorForClusterNodePools(c *v1alpha1.CassandraCluster, nodePoolNames ...string) (labels.Selector, error) {
+	selector, err := SelectorForCluster(c)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to create base selector")
+	}
+	nodePoolNameExistsRequirement, err := labels.NewRequirement(
+		v1alpha1.CassandraNodePoolNameLabel,
+		selection.Exists,
+		nil,
+	)
```

Comment:
Flatten to a single line

Reply:
Also why is this being set to nil? Doesn't make sense to me? 🤔

Reply:
`Exists` selections can not have a value.